src/lib/UniversalDiscovery/UniversalDiscoveryProvider.php

```diff
@@ -0,0 +1,346 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\UniversalDiscovery;
+
+use eZ\Publish\API\Repository\BookmarkService;
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\SearchService;
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Search\SearchHit;
+use eZ\Publish\API\Repository\Values\User\Limitation;
+use EzSystems\EzPlatformAdminUi\Permission\LookupLimitationsTransformer;
+use EzSystems\EzPlatformAdminUi\Permission\PermissionCheckerInterface;
+use EzSystems\EzPlatformAdminUi\QueryType\LocationPathQueryType;
+use EzSystems\EzPlatformRest\Output\Visitor;
+use EzSystems\EzPlatformRest\Server\Values\Version;
+
+class UniversalDiscoveryProvider implements Provider
+{
+    private const COLUMNS_NUMBER = 4;
+
+    /** @var \eZ\Publish\API\Repository\LocationService */
+    private $locationService;
+
+    /** @var \eZ\Publish\API\Repository\ContentTypeService */
+    private $contentTypeService;
+
+    /** @var \eZ\Publish\API\Repository\SearchService */
+    private $searchService;
+
+    /** @var \EzSystems\EzPlatformRest\Output\Visitor */
+    private $visitor;
+
+    /** @var \eZ\Publish\API\Repository\BookmarkService */
+    private $bookmarkService;
+
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    private $contentService;
+
+    /** @var \EzSystems\EzPlatformAdminUi\Permission\PermissionCheckerInterface */
+    private $permissionChecker;
+
+    /** @var \EzSystems\EzPlatformAdminUi\Permission\LookupLimitationsTransformer */
+    private $lookupLimitationsTransformer;
+
+    /** @var \EzSystems\EzPlatformAdminUi\QueryType\LocationPathQueryType */
+    private $locationPathQueryType;
+
+    private $sortClauseClassMap = [
+        self::SORT_CLAUSE_DATE_PUBLISHED => Query\SortClause\DatePublished::class,
+        self::SORT_CLAUSE_CONTENT_NAME => Query\SortClause\ContentName::class,
+    ];
+
+    private $availableSortOrder = [
```

Comment:
i think that if those would be keys it would look better than calling availableSortOrder[0] later

Reply:
I don't need `$availableSortOrder[0]` at all. Fixed.